app/src/main/java/io/apicurio/registry/storage/RegistryStorage.java

```diff
@@ -62,7 +71,13 @@
      * @throws RegistryStorageException
      */
     public Map<String, String> updateArtifact(String artifactId, String content) throws ArtifactNotFoundException, RegistryStorageException;
-    
+
+    /**
+     * Get all artifact ids.
+     * @return all artifact ids
+     */
+    public Set<String> getArtifactIds(/* TODO -- filter? */);
```

Comment:
Right, this is needed as well.  I don't think a filter is necessary.  I think all browse/search functionality should be done in a separate application/API that separately indexes the content from the same data store.

Reply:
I left this "filter" so that we could perhaps, if needed, filter out non-Confluent artifacts.
e.g. any non-Avro artifact / content

As with no filter, we would return **all** artifacts, not really sure we want that for ccompat ...?